### ISSUE TITLE
* aws_s3_bucket.bucket: unexpected EOF

### ISSUE THREAD

**Issue by tomjones2a** [terraform-providers/terraform-provider-aws#3325]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.11.3
+ provider.aws v1.9.0

### Affected Resource(s)
Please list the resources as a list, for example:
aws_s3_bucket

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
 vi main.tf     
provider "aws" {
 region                  = "us-east-1"
 profile                 = "saml"
}

resource "aws_s3_bucket" "bucket" {
  bucket =  "pruvpcaws016-testbucket2"

  tags {
    Purpose = "Hostname Tagging"
  }
}

```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
S3 bucket created.

### Actual Behavior
S3 bucket created but TEFFAFORM CRASH
* aws_s3_bucket.bucket: unexpected EOF


### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
2018-02-11T20:46:33.087-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: 2018/02/11 20:46:33 [ERROR] WaitForState exceeded refresh grace period
2018-02-11T20:46:33.089-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: panic: interface conversion: interface {} is nil, not *s3.GetBucketLocationOutput
2018-02-11T20:46:33.089-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:
2018-02-11T20:46:33.089-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: goroutine 151 [running]:
2018-02-11T20:46:33.089-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsS3BucketRead(0xc420499a40, 0x25663c0, 0xc42007ca00, 0x0, 0xc42036d818)



### References

**Comment by bflad** [comment]
@tomjones2a if you have the next line of the debug output above (should have filename and line number), it would be super helpful to confirm any bug fix. Thanks!

**Comment by tomjones2a** [comment]
* aws_s3_bucket.bucket: unexpected EOF
2018/02/12 12:34:29 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_s3_bucket.bucket: unexpected EOF
2018/02/12 12:34:29 [TRACE] [walkApply] Exiting eval tree: aws_s3_bucket.bucket
2018/02/12 12:34:29 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
                                                                                                                                     4031,38       99%
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc42068fa40, 0xc4205e1c20, 0xc4207641a0, 0x25663c0, 0xc420198500, 0x1, 0xc420142e10, 0xc42060d800)
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:227 +0x364
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc4203b9c00, 0xc4205e1bd0, 0xc4205e1c20, 0xc4207641a0, 0x7fa3951846c8, 0x0, 0x18)
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:283 +0xa4
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc4206e4b60, 0xc4206f58c0, 0xc4202d2e40, 0x0, 0x0)
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:527 +0x57
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: reflect.Value.call(0xc4201ae9c0, 0xc42000d980, 0x13, 0x296791e, 0x4, 0xc420a8bf20, 0x3, 0x3, 0x403a23, 0xc42002e380, ...)
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:   /usr/local/go/src/reflect/value.go:434 +0x905
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: reflect.Value.Call(0xc4201ae9c0, 0xc42000d980, 0x13, 0xc42073ff20, 0x3, 0x3, 0xc42073ff20, 0xc42073ff28, 0xc42032e5c0)
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:   /usr/local/go/src/reflect/value.go:302 +0xa4
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: net/rpc.(*service).call(0xc420804380, 0xc420066370, 0xc4202e8ab0, 0xc42007be80, 0xc4206e5180, 0x20d1400, 0xc4206f58c0, 0x16, 0x20d1440, 0xc4202d2e40, ...)
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:   /usr/local/go/src/net/rpc/server.go:381 +0x142
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: created by net/rpc.(*Server).ServeCodec
2018-02-12T12:34:29.135-0500 [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4:   /usr/local/go/src/net/rpc/server.go:475 +0x36b
2018/02/12 12:34:29 [TRACE] root: eval: *terraform.EvalWriteState
2018-02-12T12:34:29.137-0500 [DEBUG] plugin: plugin process exited: path=/.../.../.../.../.../.../terraform/.terraform/plugins/linux_amd64/terraform-provider-aws_v1.9.0_x4
2018/02/12 12:34:29 [TRACE] root: eval: *terraform.EvalApplyProvisioners
2018/02/12 12:34:29 [TRACE] root: eval: *terraform.EvalIf
2018/02/12 12:34:29 [TRACE] root: eval: *terraform.EvalWriteState
2018/02/12 12:34:29 [TRACE] root: eval: *terraform.EvalWriteDiff
2018/02/12 12:34:29 [TRACE] root: eval: *terraform.EvalApplyPost
2018/02/12 12:34:29 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* aws_s3_bucket.bucket: unexpected EOF
2018/02/12 12:34:29 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

**Comment by tomjones2a** [comment]
Mistakenly closed the issue I believe.  Sorry this is my first issue.

**Comment by bflad** [comment]
No worries, we're glad for the report so we can fix it. 😄  The fix for the original crash mentioned above has been merged into master and will be released in v1.10.0 of the AWS provider, likely next week. If you're seeing a different crash somewhere (not related to read timeouts with `aws_s3_bucket`), let's open a separate ticket. Thanks! 👍

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
